### PR TITLE
Fix1

### PR DESCRIPTION
**Pull request by fsx950223**
# Description

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py [hunk 1]
```diff
@@ -72,15 +72,31 @@ def test_get_config():
     assert config["start_averaging"] == 0
 
 
-def test_assign_batchnorm():
+def test_mixed_precision():
+    tf.keras.mixed_precision.set_global_policy("mixed_float16")
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py:76]: We have a pytest fixture

https://github.com/tensorflow/addons/blob/08767330c70ccd0974beaf3bb804dc565e9c9406/tensorflow_addons/utils/test_utils.py#L111

**Review by WindQAQ — COMMENTED** [review]
LGTM. I do think we should have some mixin classes to delegate those as we have many optimizer wrappers.

https://github.com/tensorflow/tensorflow/blob/v2.4.0/tensorflow/python/keras/mixed_precision/loss_scale_optimizer.py#L65-L180
https://github.com/tensorflow/tensorflow/blob/v2.4.0/tensorflow/python/keras/mixed_precision/loss_scale_optimizer.py#L816-L922

**Comment by fsx950223** [comment]
> LGTM. I do think we should have some mixin classes to delegate those as we have many optimizer wrappers.
> 
> https://github.com/tensorflow/tensorflow/blob/v2.4.0/tensorflow/python/keras/mixed_precision/loss_scale_optimizer.py#L65-L180
> https://github.com/tensorflow/tensorflow/blob/v2.4.0/tensorflow/python/keras/mixed_precision/loss_scale_optimizer.py#L816-L922

Split mixin classes in another PR.

File: tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py [hunk 2]
```diff
@@ -80,7 +81,6 @@ def test_assign_batchnorm():
     model.add(tf.keras.layers.Dense(16, activation="relu"))
     model.add(tf.keras.layers.BatchNormalization())
     model.add(tf.keras.layers.Dense(1))
-
     opt = SWA(tf.keras.optimizers.SGD())
     model.compile(optimizer=opt, loss="mean_squared_error")
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py:85]: The PR is caused by the fact that some callbacks access `model.optimizer.lr`. Does it get tested in this unittest? Maybe have another test instead of this one?

Comment (fsx950223) [reply tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py:85]: Oh, It's my mistake

File: tensorflow_addons/optimizers/average_wrapper.py [hunk 3]
```diff
@@ -38,7 +38,8 @@ def __init__(
             )
 
         self._optimizer = optimizer
-        self._track_trackable(self._optimizer, "awg_optimizer")
+        self._hyper.update(optimizer._hyper)
+        self._track_trackable(self._optimizer, "swg_optimizer")
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/average_wrapper.py:42]: Emm, this is the `AverageOptimizerWrapper` class so what do swg and awg stand for?

File: tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py (hunk 2 above)
Comment (bhack) [reply tensorflow_addons/optimizers/tests/stochastic_weight_averaging_test.py:101]: We are evaluating if `lr` is still valid at https://github.com/tensorflow/tensorflow/pull/44295

**Review by bhack — COMMENTED** [review]
We are evaluating if `lr` is still valid at https://github.com/tensorflow/tensorflow/pull/44295

**Comment by bhack** [comment]
What Is the status of this?

**Comment by fsx950223** [comment]
The PR remove unnecessary warnings which are created by weights can't be found, change to trainable_weights fix it.
@bhack
